Integration Tests are failing (Minishift, self healing)

Note: Until #3166 is implemented, I'll post this manually!

Integration tests have failed.

* Date Time: 2021-02-22 04:00
* Link to run: https://github.com/keptn/keptn/actions/runs/587804865
* Branch: master
* Commit:   af49f6d  
* Last committer: @Kirdock 

| Target Platform        |  Keptn Install    |  Keptn Auth  | Linking Stages   |   Quality Gates Standalone  |  Self Healing  |  Delivery Assistant  | Continuous Delivery |
| ---------------------- | ----------------- | ------------ | ---------------- | --------------------------- | -------------- | -------------------- | ------------------- |
| kubernetes/GKE 1.15 | success | success | success | success | success | success | success |
| kubernetes/GKE 1.17 | success | success | success | success | success | success | success |
| kubernetes/k3s-on-GHA v1.16.15+k3s1 | success | success | success | success | success | skipped | skipped |
| kubernetes/k3s-on-GHA v1.19.5+k3s1 | success | success | success | success | success | skipped | skipped |
| openshift/minishift-on-GHA 3.11 | success | success | success | skipped | **failure** | skipped | skipped |

Seems to be an error in openshift/minishift self-healing test: https://github.com/keptn/keptn/runs/1948759970?check_suite_focus=true#step:40:31

```
Error: [2021-02-22 03:13:41] ERROR: Checking .source, expected 'remediation-service', got 'null' ❌
Error: Process completed with exit code 1.
```

--- a/installer/manifests/keptn/charts/control-plane/templates/continuous-operations.yaml
+++ b/installer/manifests/keptn/charts/control-plane/templates/continuous-operations.yaml
@@ -79,7 +79,7 @@ spec:
           - name: PUBSUB_URL
             value: 'nats://keptn-nats-cluster'
           - name: PUBSUB_TOPIC
-            value: 'sh.keptn.>'
+            value: 'sh.keptn.event.problem.open,sh.keptn.event.evaluation.finished,sh.keptn.event.action.finished'
           - name: PUBSUB_RECIPIENT
             value: '127.0.0.1'
       serviceAccountName: keptn-default

--- a/remediation-service/handler/handler.go
+++ b/remediation-service/handler/handler.go
@@ -49,7 +49,6 @@ func NewHandler(event cloudevents.Event) (Handler, error) {
 		return nil, err
 	}
 
-	keptnHandler.Logger.Debug("Received event for shkeptncontext:" + shkeptncontext)
 	remediationHandler := &RemediationHandler{
 		Keptn:           keptnHandler,
 		RemediationRepo: &db.RemediationMongoDBRepo{},

--- a/remediation-service/main.go
+++ b/remediation-service/main.go
@@ -5,7 +5,6 @@ import (
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 	"github.com/kelseyhightower/envconfig"
 	keptnapi "github.com/keptn/go-utils/pkg/api/utils"
-	keptncommon "github.com/keptn/go-utils/pkg/lib/keptn"
 	"github.com/keptn/keptn/remediation-service/handler"
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	"log"
@@ -48,9 +47,6 @@ func _main(args []string, env envConfig) int {
 func gotEvent(ctx context.Context, event cloudevents.Event) error {
 	var shkeptncontext string
 	event.Context.ExtensionAs("shkeptncontext", &shkeptncontext)
-
-	logger := keptncommon.NewLogger(shkeptncontext, event.Context.GetID(), "remediation-service")
-	logger.Debug("Received event for shkeptncontext:" + shkeptncontext)
 
 	handler, err := handler.NewHandler(event)
 	if err != nil {

--- a/shipyard-controller/deploy/service.yaml
+++ b/shipyard-controller/deploy/service.yaml
@@ -87,7 +87,7 @@ spec:
           - name: PUBSUB_URL
             value: 'nats://keptn-nats-cluster'
           - name: PUBSUB_TOPIC
-            value: 'sh.keptn.>'
+            value: 'sh.keptn.event.problem.open,sh.keptn.event.evaluation.finished,sh.keptn.event.action.finished'
           - name: PUBSUB_RECIPIENT
             value: '127.0.0.1'
           - name: PUBSUB_RECIPIENT_PATH

--- a/test/test_self_healing.sh
+++ b/test/test_self_healing.sh
@@ -66,7 +66,7 @@ fi
 
 echo "Sending problem.open event"
 keptn_context_id=$(send_event_json ./test/assets/self_healing_problem_open_event.json)
-sleep 10
+sleep 15
 
 #response=$(curl -X GET "${KEPTN_ENDPOINT}/mongodb-datastore/event?project=${PROJECT}&type=sh.keptn.event.remediation.finished&keptnContext=${keptn_context_id}" -H  "accept: application/json" -H  "x-token: ${KEPTN_API_TOKEN}" -k 2>/dev/null | jq -r '.events[0]')
 response=$(get_keptn_event $PROJECT $keptn_context_id sh.keptn.event.remediation.finished $KEPTN_ENDPOINT $KEPTN_API_TOKEN)

--- a/test/test_self_healing.sh
+++ b/test/test_self_healing.sh
@@ -15,7 +15,7 @@ function print_logs {
   kubectl -n ${KEPTN_NAMESPACE} logs svc/remediation-service -c remediation-service
 }
 
-trap print_logs ERR EXIT
+trap print_logs EXIT
 
 ########################################################################################################################
 # Pre-requisites

